pkg/metrics/config.go

```diff
@@ -70,5 +74,9 @@ func NewObservabilityConfigFromConfigMap(configMap *corev1.ConfigMap) (*Observab
 		oc.RequestMetricsBackend = mb
 	}
 
+	if prof, ok := configMap.Data["profiling.enable"]; ok {
+		oc.EnableProfiling = strings.ToLower(prof) == "true"
+	}
```

Comment:
One question is whether we want to use the same flag or we want a separate flag for queue-proxy because changing the flag at runtime causes the application to be redeployed.